webapp/src/app/components/widgets/event-widget/event-registry/show-modal-choice.ts

```diff
@@ -0,0 +1,78 @@
+import { Label } from '../../../../models/events';
+import { EntityAttributes } from '../../../../services/phaser/entities/cc-entity';
+import { AbstractEvent, EventType } from './abstract-event';
+
+export interface ShowModalChoiceData extends EventType {
+    [key: number]: AbstractEvent<any>[];
+
+    text: Label;
+    options: {
+        label: Label;
+    }[];
+}
+
+export class ShowModalChoice extends AbstractEvent<ShowModalChoiceData> {
+    private attributes: EntityAttributes = {
+        text: {
+            type: 'LangLabel',
+            description: 'Text of modal dialog',
+			hs: true
+        },
+        options: {
+            type: 'ModalChoiceOptions',
+            description: 'All the options of the modal dialog',
+			C2: true
+        }
+    };
+
+    getAttributes(): EntityAttributes {
+        return this.attributes;    
+    }
+
+    update() {
+        this.children = [];
+		this.info = this.combineStrings(
+			this.getTypeString('#7ea3ff'),
+			this.getPropString('text', this.getProcessedText(this.data.text))
+		);
+		
+		this.data.options.forEach((option, index) => {
+			this.children[index] = {
+				title: this.getColoredString('Choice: ', '#838383') + this.getProcessedText(option.label),
+				events: this.data[index] || [],
+				draggable: false
+			};
+		});
+    }
+
+    override export(): ShowModalChoiceData {
+        const out: ShowModalChoiceData = {
+            type: this.data.type,
+            text: this.data.text,
+            options: this.data.options,
+        };
+		this.children.forEach((child, index) => {
+			if (!child.events) {
+				console.error('wtf', this);
+			}
+			out[index] = child.events.map(v => v.export());
+		});
+		
+		return JSON.parse(JSON.stringify(out));
```

Comment:
use `Helper.copy` instead to preserve the type

Reply:
That was how SHOW_CHOICE did it, so I just did it the same way (same with the 3 above things, just don't want to repeat the message), and I just fixed all that.